src/cljam/io/bam/decoder.clj

```diff
@@ -129,9 +156,14 @@
          seq         (decode-seq (lsb/read-bytes buffer (quot (inc l-seq) 2)) l-seq)
          qual        (decode-qual (lsb/read-bytes buffer l-seq))
          rest        (lsb/read-bytes buffer (options-size (alength ^bytes (:data block)) l-read-name n-cigar-op l-seq))
-         options     (decode-options rest)]
+         options     (decode-options rest)
+         [cigar* options*]
+         (if (and (contains? (first options) :CG)
```

Comment:
Is `CG` tag always appears at the first? I think the spec saids nothing about the position of the tag.

```suggestion
         (if (and (some :CG options)
```